input/system/azure/system.go

```diff
@@ -0,0 +1,302 @@
+package azure
+
+import (
+	"context"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	"github.com/Azure/azure-sdk-for-go/sdk/monitor/azquery"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/cosmosforpostgresql/armcosmosforpostgresql"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/postgresql/armpostgresqlflexibleservers/v4"
+	"github.com/pganalyze/collector/state"
+	"github.com/pganalyze/collector/util"
+)
+
+// GetSystemState - Gets system information about an Azure instance
+func GetSystemState(ctx context.Context, server *state.Server, logger *util.Logger) (system state.SystemState) {
+	config := server.Config
+	system.Info.Type = state.AzureDatabaseSystem
+	if config.AzureSubscriptionID == "" {
+		server.SelfTest.MarkCollectionAspectWarning(state.CollectionAspectSystemStats, "unable to collect system stats")
+		server.SelfTest.HintCollectionAspect(state.CollectionAspectSystemStats, "Config value azure_subscription_id is required to collect system stats.")
+		return
+	}
+
+	credential, err := getAzureCredential(config)
+	if err != nil {
+		server.SelfTest.MarkCollectionAspectError(state.CollectionAspectSystemStats, "error getting credential: %v", err)
+		server.SelfTest.HintCollectionAspect(state.CollectionAspectSystemStats, "Make sure the managed identity is assigned to the collector VM.")
+		logger.PrintError("Azure/System: Encountered error getting credential: %v\n", err)
+		return
+	}
+
+	var resourceID string
+
+	// Server info: Flexible Server
```

Comment:
Since we don't know the resource type (if Flexible server or Cosmos DB) at this point, we search in Flexible Servers first, then search in Cosmos DB.

Also, this is not getting the specific server info, but getting the all servers in that subscription. It's a bit unnecessary step (e.g. we could have asked all the needed information as config vars to avoid this), but we also want to use this pattern so that we could potentially do this at the collector level v.s. doing this over and over in each server. For example, I can make one list API request and obtain all servers info that the collector is monitoring.
(Lukas mentioned that it's the way we want to do in the future)